src/service/viewport-impl.js

```diff
@@ -405,9 +405,10 @@ export class Viewport {
    * @return {!Promise<?../layout-rect.LayoutRectDef>}}
    */
   getElementRectAsync(el) {
-    return this.vsync_.measurePromise(() => {
-      return this.getElementRect_(el);
-    });
+    return this.binding_.getElementRectAsync(el) ||
```

Comment:
@lannka I prefer to leave this as it is, at least in the PR. 
I can move the logic to each binding class, but if you take a look at our current viewport `#getLayoutRect` method, it takes care of element position inside FIE as well. If we are to override `getLayoutRectAsync` in each binding class, we may also need to move FIE logic to each binding class as well. Which doesn't sound reasonable to me.

Maybe we can think of a better way to share logic in another PR.